### PR TITLE
tropopause - add tropopause_twmo and validation test.

### PR DESCRIPTION
**Pull request by odiazib**

### CODE REVIEW

File: src/mam4xx/tropopause.hpp [hunk 1]
```diff
@@ -1,22 +1,19 @@
 #ifndef MAM4XX_TROPOPAUSE_HPP
 #define MAM4XX_TROPOPAUSE_HPP
 
-#include <haero/math.hpp>
-#include <mam4xx/aero_config.hpp>
-
 #include <fstream>
```
Comment (jeff-cohere) [inline src/mam4xx/tropopause.hpp:4]: Are `fstream` and `iostream` needed? I haven't looked closely, but I don't notice any calls to C++ streams here.

**Review by jeff-cohere — APPROVED** [review]
Looks good! Just left one minor comment. Thanks, Oscar.

**Review by mjs271 — APPROVED** [review]

File: src/mam4xx/tropopause.hpp [hunk 2]
```diff
@@ -208,6 +205,43 @@ void twmo(const ConstColumnView &temp1d, const ConstColumnView &pmid1d,
...
+void tropopause_twmo(const ConstColumnView &pmid, const ConstColumnView &pint,
+                     const ConstColumnView &temp, const ConstColumnView &zm,
+                     const ConstColumnView &zi, int &tropLev) {
+  // BAD CONSTANT
+  constexpr Real gam =
+      -0.002; //       ! lapse rate to indicate tropopause [K/m]
+  constexpr Real plimu =
+      45000; //       ! upper limit of tropopause pressure [Pa]
+  constexpr Real pliml =
+      7500; //        !lower limit of tropopause pressure [Pa]
```
Comment (jaelynlitz) [inline src/mam4xx/tropopause.hpp:221]:
```suggestion
  constexpr Real gam =
      -0.002; // lapse rate to indicate tropopause [K/m]
  constexpr Real plimu =
      45000; // upper limit of tropopause pressure [Pa]
  constexpr Real pliml =
      7500; // lower limit of tropopause pressure [Pa]
```

File: src/mam4xx/tropopause.hpp [hunk 3]
```diff
@@ -208,6 +205,43 @@ void twmo(const ConstColumnView &temp1d, const ConstColumnView &pmid1d,
...
+
+  // if successful, store of the results and find the level and temperature.
+  if (tP > 0) {
+
+    // Find the associated level.
+    for (int kk = pver - 1; kk > 1; --kk) {
+      if (tP >= pint(kk)) {
+        tropLev = kk;
+        // tropLevValp1 = kk + 1;
+        // tropLevValm1 = kk - 1;
```
Comment (jaelynlitz) [inline src/mam4xx/tropopause.hpp:235]: Should these comments be deleted? Same on line 240

**Review by jaelynlitz — COMMENTED** [review]
LGTM thanks @odiazib!

**Review by jaelynlitz — APPROVED** [review]

File: src/mam4xx/tropopause.hpp (hunk 2 above)
Comment (jaelynlitz) [reply src/mam4xx/tropopause.hpp:221]: just making this consistent with your changes to the comments in `twmo` 😄

File: src/mam4xx/tropopause.hpp (hunk 1 above)
Comment (odiazib) [reply src/mam4xx/tropopause.hpp:4]: I do not think so. I will remove it.

**Pull request merged**